pkg/prometheus/agent_http_test.go

```diff
@@ -0,0 +1,268 @@
+package prometheus
```

Comment:
Loving these tests 😍 

I'm a fan of tests that assert against plain-text fixtures (e.g. JSON as string) but I understand it comes with trade-offs. The structure and assertions look great nonetheless.